build.go

```diff
@@ -131,37 +146,55 @@ func buildBlock(sb io.StringWriter, parent *Directive, block Directives, depth i
 			directive := Enquote(stmt.Directive)
 			_, _ = sb.WriteString(directive)
 
-			// special handling for if statements
-			if directive == "if" {
-				_, _ = sb.WriteString(" (")
-				for i, arg := range stmt.Args {
-					if i > 0 {
-						_, _ = sb.WriteString(" ")
+			if options.ExternalBuilds != nil {
+				extDirectivesMap := make(map[string]ExternalBuilder)
+				for _, ext := range options.ExternalBuilds {
+					directives := ext.RegisterExternalBuilder()
+					for _, d := range directives {
+						extDirectivesMap[d] = ext
+					}
+
+					if ext, ok := extDirectivesMap[directive]; ok {
+						_, _ = sb.WriteString(" ") // space between directives and arguments
+						if err := ext.Build(sb, stmt); err != nil {
+							log.Printf("Failed to write externaller block: %v", err)
```

Comment:
I don't think it's appropriate to log here. If this method returns an error then I think it needs to be returned to the caller, which is not part of this function's signature... Either we need to change the function to return it or design the interface in a way that does not enable returning an error.

Reply:
I changed the interface and used the one that we talk in Slack.